include/tbb/concurrent_hash_map.h

```diff
@@ -664,7 +664,9 @@ class concurrent_hash_map : protected internal::hash_map_base {
         return create_node(allocator, std::piecewise_construct,
                            std::forward_as_tuple(key), std::forward_as_tuple());
 #else
-        T obj; // Use of temporary object in impossible, because create_node takes non-const reference
+        // Use of a temporary object is impossible, because create_node takes a non-const reference.
+        // copy-initialization is possible because T is already required to be CopyConstructible.
+        T obj = T();
```

Comment:
Why not `T obj()`?

Reply:
That would declare `obj` as a function taking `void` and returning `T`, because the compiler is obliged to prioritise this interpretation of the ambiguity. With C++11 you can avoid it by using `T obj{};`, which would not depend on the presence of a copy constructor for class object types (most/all compilers perform the copy elision and so only insist to see the declaration but do not require a definition), until the problem goes away in C++17 (copy constructor no longer needed, compiler is obliged to elide the copy, and you're not even supposed to use that term anymore which seems quite silly to me).

I should probably have differentiated based on language version, and then some code might work from C++11 instead of only from C++17, so I would recommend that you amend this.

Reply:
Surely, you are right about [most vexing parse](https://en.wikipedia.org/wiki/Most_vexing_parse). Ignore my irrelevant comment. (Writing `()` I thought about `{}` that is not applicable for pre-C++11.)